sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/fl_scatter_gather.py

```diff
@@ -0,0 +1,771 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+from typing import Dict, Union, List, Optional
+import re
+
+from azure.ai.ml import Output
+from azure.ai.ml.constants._common import AssetTypes
+from azure.ai.ml._schema import PathAwareSchema
+from azure.ai.ml._schema.pipeline.control_flow_job import FLScatterGatherSchema
+from azure.ai.ml.entities._builders.control_flow_node import ControlFlowNode
+from azure.ai.ml.entities._job.pipeline._io.mixin import NodeIOMixin
+from azure.ai.ml.entities._util import convert_ordered_dict_to_dict
+from azure.ai.ml.constants import JobType
+
+from azure.ai.ml.entities._assets.federated_learning_silo import FederatedLearningSilo
+from azure.ai.ml.entities._component.component import Component
+from azure.ai.ml.entities._validation import MutableValidationResult
+from .subcomponents import aggregate_output
+
+# TODO 2293610: add support for more types of outputs besides uri_folder and mltable
+# Likely types that ought to be mergeable: string, int, uri_file
+MERGE_COMPONENT_MAPPING = {
+    "mltable": aggregate_output,
+    "uri_folder": aggregate_output,
+}
+
+
+ANCHORABLE_OUTPUT_TYPES = {AssetTypes.MLTABLE, AssetTypes.URI_FOLDER}
+
+ANCHORING_PATH_ROOT = "root"
+
+
+# big TODO: For some reason, surfacing this file in  __init__.py causes
+#  a circular import exception on the first attempted import
+# In notebooks, the second import succeeds, but then causes a silent failure where the
+# MLDesigner component created by the subcomponents.aggregate_output function
+# will produce a ComponentExecutor object instead of the actual component.
+# TODO 2293541: Add telemetry of some sort
+# pylint: disable=too-many-instance-attributes
+class FLScatterGather(ControlFlowNode, NodeIOMixin):
```

Comment:
Similar to schema comment, is there anything special for the inheritance from `ControlFlowNode`?

Reply:
No, we just discussed it and felt that it was the most appropriate node to inherit from a human-logic standpoint.